docs/topics/request-response.rst

```diff
@@ -289,6 +304,128 @@ Example:
 
+What callbacks can return and yield
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Callback functions can return or yield the following types of objects:
+
+**Items**: Any :ref:`item object <topics-items>` (dictionaries, :class:`~scrapy.Item`
+instances, etc.) that will be processed by the :ref:`item pipeline <topics-item-pipeline>`.
+
+**Requests**: :class:`~scrapy.Request` objects (or subclasses like
+:class:`~scrapy.FormRequest`) that will be scheduled for download.
+
+**None**: Returning or yielding ``None`` is valid and results in no action
+being taken for that particular return/yield. This is useful when you want
+to conditionally process responses.
+
+**Iterables**: Any iterable containing a mix of the above types. You can return
+   or yield lists, generators, or any other iterable containing items and requests.
+
+**Async generators**: Callbacks can be defined as ``async def`` functions and
+use ``yield`` to produce items and requests asynchronously.
+
+Examples:
+
+.. code-block:: python
+
+    def parse(self, response):
+        # Return a single item
+        yield {"title": response.css("h1::text").get()}
+        
+        # Return a single request
+        yield scrapy.Request(response.urljoin("/next-page"), self.parse)
+        
+        # Return None (no-op)
+        if not response.css("h1"):
+            return None
+            
+        # Return multiple items and requests
+        for link in response.css("a::attr(href)"):
+            yield scrapy.Request(response.urljoin(link.get()), self.parse)
+            yield {"link": link.get()}
+            
+        # Return/yield an iterable (list, generator, etc.)
+        items = [{"title": "Item 1"}, {"title": "Item 2"}]
+        requests = [scrapy.Request(url, self.parse) for url in ["/page1", "/page2"]]
+        yield items  # Yields the entire list
+        yield requests  # Yields the entire list of requests
+
+    # Async callback example
+    async def parse_async(self, response):
+        # Process response asynchronously
+        title = response.css("h1::text").get()
+        if title:
+            yield {"title": title}
+            
+        # Follow links asynchronously
+        for link in response.css("a::attr(href)"):
+            yield scrapy.Request(response.urljoin(link.get()), self.parse_async)
+
+Valid callback types
+~~~~~~~~~~~~~~~~~~~~
+
+Callbacks can be:
+
+**Spider methods**: The most common approach. The callback must be a method
+of the spider class.
+
+.. code-block:: python
+
+    class MySpider(scrapy.Spider):
+        def parse(self, response):
+            pass
+            
+        def parse_item(self, response):
+            pass
+
+**String references**: You can pass a string containing the method name.
+This is useful for dynamic callback selection.
+
+.. code-block:: python
+
+    def get_callback(self, url):
+        if "/products/" in url:
+            return "parse_product"
+        return "parse_category"
```

Comment:
I'm not sure what's the full context of this `get_callback` method, but it doesn't look correct: a Request's callback must be callable (https://github.com/scrapy/scrapy/blob/2.13.3/scrapy/http/request/__init__.py#L145-L148)